experimental/adaptive-card-prompt-sample/javascript_nodejs/adaptiveCardPrompt.js

```diff
@@ -0,0 +1,222 @@
+// Note: This is an edit of the compiled version of: 
+// https://github.com/mdrichardson/botbuilder-js/blob/adaptiveCardPrompt/libraries/botbuilder-dialogs/src/prompts/adaptiveCardPrompt.ts
+
+const { Dialog } = require("botbuilder-dialogs");
+const { InputHints } = require("botbuilder");
+/**
+ * Waits for Adaptive Card Input to be received.
+ *
+ * @remarks
+ * This prompt is similar to ActivityPrompt but provides features specific to Adaptive Cards:
+ *   * Card can be passed in constructor or as prompt/reprompt activity attachment
+ *   * Includes validation for specified required input fields
+ *   * Displays custom message if user replies via text and not card input
+ *   * Ensures input is only valid if it comes from the appropriate card (not one shown previous to prompt)
+ * DO NOT USE WITH CHANNELS THAT DON'T SUPPORT ADAPTIVE CARDS
+ */
+class AdaptiveCardPrompt extends Dialog {
+    /**
+     * Creates a new AdaptiveCardPrompt instance
+     * @param dialogId Unique ID of the dialog within its parent `DialogSet` or `ComponentDialog`.
+     * @param validator (optional) Validator that will be called each time a new activity is received. Validator should handle error messages on failures.
+     * @param settings (optional) Additional options for AdaptiveCardPrompt behavior
+     */
+    constructor(dialogId, validator, settings) {
+        super(dialogId);
+        this.usesCustomPromptId = false;
+        // Necessary for when this compiles to js since strictPropertyInitialization is false/unset in tsconfig
+        settings = typeof (settings) === 'object' ? settings : {};
+        this.validator = validator;
+        this._inputFailMessage = settings.inputFailMessage || 'Please fill out the Adaptive Card';
+        this._requiredInputIds = settings.requiredInputIds || [];
+        this._missingRequiredInputsMessage = settings.missingRequiredInputsMessage || 'The following inputs are required';
+        this._attemptsBeforeCardRedisplayed = settings.attemptsBeforeCardRedisplayed || 3;
+        this._card = settings.card;
+        if (settings.promptId) {
+            this._promptId = settings.promptId;
+            this.usesCustomPromptId = true;
+        }
+    }
+    get inputFailMessage() {
+        return this._inputFailMessage;
+    }
+    set inputFailMessage(message) {
+        this._inputFailMessage = message;
+    }
+    get requiredInputIds() {
+        return this._requiredInputIds;
+    }
+    set requiredInputIds(ids) {
+        this._requiredInputIds = ids;
+    }
+    get missingRequiredInputsMessage() {
+        return this._missingRequiredInputsMessage;
+    }
+    set missingRequiredInputsMessage(message) {
+        this._missingRequiredInputsMessage = message;
+    }
+    get attemptsBeforeCardRedisplayed() {
+        return this._attemptsBeforeCardRedisplayed;
+    }
+    set attemptsBeforeCardRedisplayed(attempts) {
+        this._attemptsBeforeCardRedisplayed = attempts;
+    }
```

Comment:
@stevengum Quick note on all reviews for `adaptiveCardPrompt.js/ts`: I plan on submitting a separate PR for those and then updating the sample with them. I think it makes more sense to have a deep-dive into them there, rather than the samples. I'll incorporate most changes before submitting the PR.

---

I really should have left a comment for this. I recall going back and forth on setters/getters vs properties and can't remember why I decided on setters/getters. Probably just a naive mistake, now.